Cython/CodeWriter.py

```diff
@@ -247,10 +248,37 @@ def visit_FuncDefNode(self, node):
         self.visit(node.body)
         self.dedent()
 
+    def visit_CFuncDefNode(self, node):
+        if 'inline' in node.modifiers:
+            return
+        if node.overridable:
+            self.startline(u'cpdef ')
+        else:
+            self.startline(u'cdef ')
+        if node.visibility != 'private':
+            self.put(node.visibility)
+            self.put(u' ')
+        if node.api:
+            self.put(u'api ')
+
+        if node.base_type:
+            self.visit(node.base_type)
+            if node.base_type.name is not None:
+                self.put(u' ')
```

Comment:
Hmm, you're making an assumption about the node type in `node.base_type` here that I'd like to avoid.
Instead, I think we should change `visit_CSimpleBaseTypeNode()` to not output the trailing space if it's not writing a name.

Reply:
Agree, but not sure if not printing the trailing space `visit_CSimpleBaseTypeNode()` would causes problems in other parts.

Reply:
Will try to revise this.

Reply:
`visit_CSimpleBaseTypeNode` is not always needs to print the ` `, otherwise it will break other things.